File: library/src/main/java/com/github/library/enums/AuthType.java
package com.github.library.enums;

public enum AuthType {

    NO_AUTH(0),
    BASIC_AUTH(1);

    private final int value;

    private AuthType(int value) {
        this.value = value;
    }

    public int getCode() {
        return this.value;
    }

    public static AuthType Parse(int value) {
        if (value == 0) {
            return NO_AUTH;
        }
        AuthType[] arr$ = values();
        for (AuthType val : arr$) {
            if (val.value == value) {
                return val;
            }
        }
        return NO_AUTH;
    }
}

File: library/src/main/java/com/github/library/enums/AuthType.kt
package com.github.library.enums

enum class AuthType(val code: Int) {

    NO_AUTH(0),
    BASIC_AUTH(1),
    BEARER_AUTH(2),
    OAUTH2_AUTH(3);

    companion object {
        fun parse(value: Int): AuthType {
            if (value == 0) {
                return NO_AUTH
            }
            val `arr$` = values()
            for (`val` in `arr$`) {
                if (`val`.code == value) {
                    return `val`
                }
            }
            return NO_AUTH
        }
    }
}
